Source/Rio.API/Controllers/ParcelLedgerController.cs

```diff
@@ -33,11 +34,22 @@ public IActionResult New([FromBody] ParcelLedgerCreateDto parcelLedgerCreateDto)
             var parcelDto = Parcel.GetByParcelNumberAsDto(_dbContext, parcelLedgerCreateDto.ParcelNumbers[0]); 
             if (parcelDto == null)
             {
-                ModelState.AddModelError("ParcelNumber", $"{parcelLedgerCreateDto.ParcelNumbers[0]} is not a valid Parcel APN.");
-                return BadRequest(ModelState);
+                if (parcelLedgerCreateDto.ParcelNumbers[0] == null)

```

Comment:
are you always passing an array of at least one here?  I would think you would need to check if you have any ParcelNumbers first

Reply:
The form just sends the inputted parcel number as a list so we can reuse the dto, so we can assume there will be one value but that the value might null.

Reply:
Function reworked per the recommendations made earlier